src/Features/Core/Portable/GenerateFromMembers/AbstractGenerateFromMembersService.cs

```diff
@@ -74,27 +74,20 @@ protected AbstractGenerateFromMembersCodeRefactoringProvider()
 
         // Can use non const fields and properties with setters in them.
         protected static bool IsWritableInstanceFieldOrProperty(ISymbol symbol)
-            => IsViableInstanceFieldOrProperty(symbol) &&
-               IsWritableFieldOrProperty(symbol);
+            => !symbol.IsStatic &&
+               (symbol is IFieldSymbol field && IsViableField(field) && !field.IsConst ||
+                symbol is IPropertySymbol property && IsViableProperty(property) && property.IsWritableInConstructor());
```

Comment:
this mix of &&'s and ||'s is confusing.  Please extract into helper and parenthesize to keep precedence clear.  In general, we prefer expressions that are uniform with &&'s or ||'s.

Reply:
I thought about this for a while and found it hard to extract common code between `IsWritableInstanceFieldOrProperty` and `IsReadableInstanceFieldOrProperty`. They use the same logic for whether the field/property is "viable" but they also need to do more checks that are still specific to each type. Basically one method would check the type and some condition, and the other one would check the type again for another condition. So I don't know what to do there.

Reply:
> IsWritableInstanceFieldOrProperty 

> IsReadableInstanceFieldOrProperty

Based on the name, it sounds like they can reuse whatever computes if this is an InstanceFieldOrProperty

Reply:
> I thought about this for a while and found it hard to extract common code

Right now the mixing of precedence operators is confusing here.  Please at least parenthesize for clarity.  Also, to prevent mixing and matching of operators, just make this a full body method.  THis is complex enough that collapsing into an expression is harder to understand.

Reply:
> Also, to prevent mixing and matching of operators, just make this a full body method. THis is complex enough that collapsing into an expression is harder to understand

Yes. But then you'd still have another method for IsWritableInstanceFieldOrProperty which would have the same structure - it needs to check writability based on whether the symbol is a field or a property. So the code reuse would actually reuse some things but duplicate other things, and introduce unnecessary type checks after we already did the same check before.